packages/nimble-components/src/wafer-map/specs/features/grid-dimension.md

```diff
@@ -0,0 +1,28 @@
+# Nimble Wafer Map Grid Dimensions
+
+## Problem Statement
+
+The Wafer Map displays dies on a canvas inside a circular border based on an internal grid which is calculated based on the data input.
+This approach is limiting the performance of the wafer and the grid dimensions can be erroneous if the displayed data is just a subset of a larger data set.
+To mitigate these problems, we will reveal three new API inputs which will allow the users to set the grid dimensions themselves, thus saving processing time and correcting inconsistencies generated by the lack of data context.
+
+## Links To Relevant Work Items and Reference Material
+
+-   [Wafer Map Spec](../README.md)
+-   [Interactive Wafer Map Improvements](https://dev.azure.com/ni/DevCentral/_wiki/wikis/Stratus/66917/Interactive-Wafer-Map-Improvements)
+
+## Implementation / Design
+
+The new attributes and properties will be described in the component definition. They will be created with change listeners which will trigger the appropriate updates in the wafer.
+
+The inputs will be validated before use. If they are undefined the previous behavior of calculating the values based on the input data will be implemented.
+
+If part of the input data is missing, the wafer will not render, and a new `validity` API will be populated with the reason for the error, `invalidGridDimensions`.
+
+## Alternative Implementations / Designs
+
+Yes.
```

Comment:
The goal of this section is to list the alternatives considered / researched for posterity and to give some discussion about why the proposed solution of the spec was selected over the alternatives.

Can you list the alternatives?

One alternative I'm curious about is that based on the description this change sounds like the goal is to enable a form of optimization. I'm assuming that means some form of benchmark or some example data sets were used that signified the need to add new APIs specific to optimization. Can you capture that information here?

Given that information, I'm curious if alternatives were considered to improve the performance of the wafer map generally? What are the performance goals we need to achieve. It is quite possible that we can leverage features of the browser to make the wafer map itself faster for large calculations. I'd like to know what alternatives were considered.

Reply:
Sorry if the document made the performance gains seem the target of this change, I tried to clarify the goals of fixing display issues with incomplete sets of data more proeminent. The improved performance is just a side effect of removing a data parsing heuristic from the update loop.

We are still researching ways to improve the performance of the wafer. A recent project saw the implementation of WebGL to render the canvas with 60 fps for a data quantity of around 3 M points as an alternative for advanced graphics capable devices. But those projects are still in research phase as we are focusing on aligning different parts of our product to create a better synergy for the end users

Reply:
> A recent project saw the implementation of WebGL to render the canvas with 60 fps for a data quantity of around 3 M points as an alternative for advanced graphics capable devices.

Is that an example of a user requirement? It sounds like a demonstration of what a browser is capable of. But just to be clear are we expecting users to need to supply 3M data points to the nimble wafermap?

What are the functional performance requirements?